app/controllers/posts_controller.rb

```diff
@@ -9,4 +9,34 @@ def show
     @post = Post.find(params[:id])
     @comments = @post.comments.all
   end
+
+  def new
+    @user = current_user
+  end
+
+  def create
+    @user = current_user
+    @post = @user.posts.new(post_params)
+
+    if @post.save
+      redirect_to user_post_path(@user, @post)
+    else
+      render :new, status: :unprocessable_entity
+    end
+  end
+
+  def like
+    @post = Post.find(params[:id])
+    @user = User.find(params[:user_id])
+    @current_user = current_user
+    @like = @current_user.likes.new(post_id: @post.id)
+    @like.save
+    redirect_to user_post_path(@user, @post)
```

Comment:
- [ ] <b>REQUIRED:</b> Also here as I mentioned in the previous comment, we only need one user which is `current_user`. and also to handle the situation in a case `likes` can't be saved. 

```ruby
 def create
      @post = Post.find(params[:post_id])
      @like = @current_user.likes.new(post_id: @post.id)

      respond_to do |format|
        if @like.save
          format.html { redirect_to @post }
        else
          format.html { redirect_to @post }
        end
      end
    end
```